source/common/tls/tls_certificate_selector_impl.cc

```diff
@@ -0,0 +1,59 @@
+#include "source/common/tls/tls_certificate_selector_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Tls {
+
+Ssl::SelectionResult
+TlsCertificateSelectorImpl::selectTlsContext(const SSL_CLIENT_HELLO* ssl_client_hello,
+                                             Ssl::CertSelectionCallbackSharedPtr cb) {
+  auto selection_ctx = cb_.lock();
```

Comment:
I think this is called syncronously within the handshake. Is it even possible for this to be null?

Reply:
oh, you are right, it should be impossible, added `IS_ENVOY_BUG` when it's null.